drivers/javascript/cursor.coffee

```diff
@@ -160,23 +161,14 @@ class IterableResult
         throw new err.ReqlDriverError "The `hasNext` command has been removed since 1.13. Use `next` instead."
 
     _next: varar 0, 1, (cb) ->
+        if cb? and typeof cb isnt "function"
+            throw new err.ReqlDriverError "First argument to `next` must be a function or undefined."
+
         fn = (cb) =>
             @_cbQueue.push cb
             @_promptNext()
 
-        if typeof cb is "function"
-            fn(cb)
-        else if cb is undefined
-            p = new Promise (resolve, reject) ->
-                cb = (err, result) ->
-                    if (err)
-                        reject(err)
-                    else
-                        resolve(result)
-                fn(cb)
-            return p
-        else
-            throw new err.ReqlDriverError "First argument to `next` must be a function or undefined."
+        return Promise.fromNode(fn).nodeify(cb)
```

Comment:
Damn, that is a nice cleanup. I need to do that in the other places we do this manually...